logstash-core/lib/logstash/settings.rb

```diff
@@ -200,25 +225,61 @@ def coerce(value)
           raise ArgumentError.new("could not coerce #{value} into a boolean")
         end
       end
+    end
 
-      def set(value)
-        coerced_value = coerce(value)
-        validate(coerced_value)
-        @value = coerce(coerced_value)
-        @value_is_set = true
-        @value
+    class Numeric < Coercible
+      def initialize(name, default=nil, strict=true)
+        super(name, ::Numeric, default, strict)
+      end
+
+      def coerce(v)
+        return v if v.is_a?(::Numeric)
```

Comment:
Instead of calling it Numeric we can call Number, but we have the same problem in the String setting.
So the only work "fix" here is calling it NumericSetting, StringSetting, etc.
Other suggestions?